Source/Plugins/bsfRenderBeast/BsRenderBeastTest.cpp

```diff
@@ -0,0 +1,50 @@
+//************************************ bs::framework - Copyright 2018 Marko Pintera **************************************//
+//*********** Licensed under the MIT license. See LICENSE.md for full terms. This notice is not to be removed. ***********//
+#include "Testing/BsConsoleTestOutput.h"
```

Comment:
I'd like to keep these non-unit tests elsewhere. Generally these things might require custom assets and might be bigger than one file as well, so it makes sense to me to dedicate a space for them. Even though this is a simple one I feel it fits into that category.

Perhaps we can add a `Tests` folder to `bsfExamples` repository? Or a custom repository.

Reply:
This also applies to `BsUtilityTest.cpp` above.

Reply:
eh i prefer tests to be close to the code of interest. otherwise they can too quickly go out of date. 
unit-tests shouldn't be too dependent on assets. unit-tests != render tests != gameplay tests and can be separate. I'd prefer to be able to add a lot more tests to validate the serialization, threading, memory, perf. and math.

Reply:
the issue with separate repos is that the most common example is a method signature changes, and then the author needs to remember to go to the separate repo, and update it there as well. Ideally the dev should just be compiling with build_tests=ON and so the tests are always compiling along with any signature changes.

Reply:
anyone compiling the source code should be running the unit-tests on their platform as well to validate.

Reply:
Makes sense. Because you are starting up the application I assumed this was doing more than just a unit test.

Reply:
ah no, starting up the application was only necessary for starting some of the modules. some of the tests involved code that had to be initialized by the modules. At the very least Platform, MemStack, MessageHandler had to be startUp(). otherwise any tests using memstack will fail obviously.